.github/workflows/ci.yml

```diff
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - 'functions/source/soci-index-generator-lambda/**'
+      - '.github/workflows/ci.yml'
+  pull_request:
+    branches: [ main ]
+    paths:
+      - 'functions/source/soci-index-generator-lambda/**'
+      - '.github/workflows/ci.yml'
+
+env:
+  GO_VERSION: '1.24'
+  GOLANGCI_LINT_VERSION: '2.1.6'
+
+jobs:
+  lint:
+    strategy:
+      matrix:
+        working_dir: ['functions/source/soci-index-generator-lambda']
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+      - uses: golangci/golangci-lint-action@v8
+        with:
+          version: v${{ env.GOLANGCI_LINT_VERSION }}
+          working-directory: ${{ matrix.working_dir }}
+
+  test:
+    name: Unit Test
+    runs-on: ubuntu-latest
+    
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
```

Comment:
Do we want to pin this to a commmit like we do in the other projects?

Reply:
is there are reason for pinning?

Reply:
Changes to the way checkout is executed(with fetch depth by default or not for example) via the actions could affect the rest of the workflow, don't see those kind of issues affecting this workflow.

LGTM

Reply:
for example, soci-snapshotter has not pinned the checkout version too https://github.com/awslabs/soci-snapshotter/blob/main/.github/workflows/build.yml#L38